0_intro_to_jax_for_ml.ipynb

```diff
@@ -0,0 +1,2081 @@
+{
```

Comment:
<p>The key to getting things like vmap is that jax functions are pure. Both of these JAX design decisions only make sense in that light (you can't have mutable rng state in pure functions, and you can't have jax pure functions mutating input arrays. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2022/pull/4/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Reply:
<p>Added pure functions section. </p>